astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,338 @@
+            values: ASC, ASZ, FTZ, HTM, IND, PDF, PNG
+
+
+        Returns
+        -------
+        None if not verbose. It downloads the observation indicated
+        If verbose returns the filename
+        """
+
+        link = self.data_aio_url + "obsno=" + observation_id
+
+        link = link + "".join("&{0}={1}".format(key, val)
+                              for key, val in kwargs.items())
+
+        if verbose:
+            log.info(link)
+
+        response = None
+        try:
+            response = self._request('GET', link, save=True, cache=True)
+        except HTTPError:
+            log.error("Unable to access URL {0}".format(link))
+
+        if response:
+            if filename is None:
+                filename = observation_id + ".tar"
+
+            log.info("Copying file to {0}...".format(filename))
+            path = os.getcwd()
+            destfile = os.path.join(path)
+            file = None
+            try:
+                os.makedirs(destfile, exist_ok=True)
+                file = open(os.path.join(destfile, filename), "w+")
+            except OSError:
+                log.error("Creation of the directory %s failed" % destfile)
+            else:
+                if verbose:
+                    log.info("Successfully created the directory %s " %
+                             destfile)
+                shutil.copy(response,
+                            os.path.join(path, filename))
+
+            if verbose:
+                log.info("Wrote {0} to {1}".format(link, filename))
+
+            return file
+
+    def get_postcard(self, observation_id, image_type="OBS_EPIC",
+                     filename=None, verbose=False):
+        """
+        Download postcards from XSA
+
+        Parameters
+        ----------
+        observation_id : string
+            id of the observation for which download the postcard, mandatory
+            The identifier of the observation we want to retrieve, regardless
+            of whether it is simple or composite.
+        image_type : string
+            image type, optional, default 'OBS_EPIC'
+            The image_type to be returned. It can be: OBS_EPIC,
+            OBS_RGS_FLUXED, OBS_RGS_FLUXED_2, OBS_RGS_FLUXED_3, OBS_EPIC_MT,
+            OBS_RGS_FLUXED_MT, OBS_OM_V, OBS_OM_B, OBS_OM_U, OBS_OM_L,
+            OBS_OM_M, OBS_OM_S, OBS_OM_W
+        filename : string
+            file name to be used to store the postcard, optional, default None
+        verbose : bool
+            optional, default 'False'
+            Flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the observation postcard indicated
+        """
+
+        params = {'RETRIEVAL_TYPE': 'POSTCARD',
+                  'OBSERVATION_ID': observation_id,
+                  'OBS_IMAGE_TYPE': image_type,
+                  'PROTOCOL': 'HTTP'}
+
+        if verbose:
+            log.info(self.data_url)
+
```

Comment:
the entire rest of this function should be replaced with
```
        local_filepath = self._request('GET', self.data_url, params, cache=True, save=True)
        return local_filepath
```

The only thing needed from the code below is the filename, which could be retrieved with something like:
```
    response = self._request('HEAD', url)
    response.raise_for_status()
    filename = re.findall('filename="(.+)"',
                                          response.headers[
                                              "Content-Disposition"])[0]
```
then 
```
shutil.move(local_filepath, filename)
```